react-components/src/components/Image360CollectionContainer/useApply360AnnotationStyling.tsx

```diff
@@ -0,0 +1,114 @@
+/*!
+ * Copyright 2024 Cognite AS
+ */
+
+import {
+  type Cognite3DViewer,
+  type Image360Collection,
+  type Image360Annotation
+} from '@cognite/reveal';
+import { Color } from 'three';
+import { image360CollectionExists } from '../../utilities/modelExists';
+import { useCallback, useEffect, useRef, useState } from 'react';
+import { type CogniteInternalId } from '@cognite/sdk';
+import { useReveal } from '../RevealCanvas/ViewerContext';
+
+export type AnnotationIdStylingGroup = {
+  assetIds: CogniteInternalId[];
+  style: Color;
+};
+
+export type ImageCollectionModelStyling = {
+  defaultStyle?: Color;
+  groups?: AnnotationIdStylingGroup[];
+};
+
+export const useApply360AnnotationStyling = (
+  imageCollection?: Image360Collection,
+  styling?: ImageCollectionModelStyling
+): void => {
+  const viewer = useReveal();
+  const [lastStyledImageAnnotations, setLastStyledImageAnnotations] = useState<
+    Image360Annotation[]
+  >([]);
+
+  const abortController = useRef(new AbortController());
+
+  const defaultStyle = styling?.defaultStyle ?? new Color(0xffffff);
+  const styleGroups = styling?.groups;
+
+  const enableAnnotationStyling = useCallback(() => {
+    if (imageCollection === undefined) return;
+    lastStyledImageAnnotations.forEach((a) => {
+      a.setColor(undefined);
+    });
+    setLastStyledImageAnnotations([]);
+    imageCollection.setDefaultAnnotationStyle({
+      color: defaultStyle,
+      visible: true
+    });
+    viewer.requestRedraw();
+  }, [viewer, defaultStyle]);
```

Comment:
So if I understand this correctly, `enableAnnotationStyling` removes styling on the last styled annotation instances, and updates the default annotation style? I don't see the correlation between the function name and what it does :monocle_face: